### PR TITLE
BUG/PERF: Fixed IntervalIndex.nbytes and itemsize

### PR DESCRIPTION
**Pull request by TomAugspurger**
Avoid materializing the ndarray of Intervals. Just get from left and right
instead.

Closes https://github.com/pandas-dev/pandas/issues/19209

### CODE REVIEW

File: pandas/core/indexes/interval.py [hunk 1]
```diff
@@ -728,6 +728,14 @@ def size(self):
         # Avoid materializing self.values
         return self.left.size
 
+    @property
+    def nbytes(self):
+        return self.left.nbytes + self.right.nbytes
+
+    @property
+    def itemsize(self):
+        return self.left.itemsize + self.right.itemsize
```
Comment (jreback) [inline pandas/core/indexes/interval.py]: this doesn’t really make sense to have 
why do u need this?

File: pandas/tests/indexes/interval/test_interval.py [hunk 2]
```diff
@@ -24,6 +24,7 @@ def name(request):
 
 class TestIntervalIndex(Base):
     _holder = IntervalIndex
+    _compat_props = ['shape', 'ndim', 'size']
```
Comment (jreback) [inline pandas/tests/indexes/interval/test_interval.py]: what is this for? we don’t use it anywhere

**Review by jreback — CHANGES_REQUESTED** [review]
maybe add an asv?

**Comment by TomAugspurger** [comment]
Sorry, was just using this to demo stuff for a talk on OSS I gave last night.

I think this will be solved by Interval being an ExtensionArray (should have that PR today).

**Comment by TomAugspurger** [comment]
Closing in favor of #20611.

**Pull request closed without merging**